scheduler/src/main/java/io/camunda/zeebe/scheduler/future/CompletableActorFuture.java

```diff
@@ -210,8 +216,23 @@ public void block(final ActorTask onCompletion) {
 
   @Override
   public void onComplete(final BiConsumer<V, Throwable> consumer) {
-    final ActorControl actorControl = ActorControl.current();
-    actorControl.runOnCompletion(this, consumer);
+    if (ActorThread.isCalledFromActorThread()) {
+      final ActorControl actorControl = ActorControl.current();
+      actorControl.runOnCompletion(this, consumer);
+    } else {
+      // We don't reject this because, this is useful for tests. But the warning is a reminder not
+      // to use this in production code.
+      Loggers.ACTOR_LOGGER.warn(
+          "No executor provided for ActorFuture#onComplete callback."
+              + " This could block the actor that completes the future."
+              + " Use onComplete(consumer, executor) instead.");
+      onComplete(consumer, Runnable::run);
+    }
+  }
+
+  @Override
+  public void onComplete(final BiConsumer<V, Throwable> consumer, final Executor executor) {
+    blockedCallbacks.add((res, error) -> executor.execute(() -> consumer.accept(res, error)));
```

Comment:
:question: Should we allow this to be called from an actor? I don't see a reason why not but maybe it'd be good to document the behavior in that case:
1. `consumer` will be called by `executor` directly
2. The calling actor is not involved at any point, i.e. `executor` and `consumer` are called outside of actor context.

Reply:
I thought it is already documented:
```
Registers a consumer, which is executed after the future was completed. The consumer is
   * executed in the provided executor.
```

Is this not clear?

Reply:
Ah my bad, I missed that you added this to the interface. All good then :+1: